jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -137,10 +137,11 @@ static RegistryAuthenticator fromAuthenticationMethod(
 
     Pattern servicePattern = Pattern.compile("service=\"(.*?)\"");
     Matcher serviceMatcher = servicePattern.matcher(authenticationMethod);
-    if (!serviceMatcher.find()) {
-      throw newRegistryAuthenticationFailedException(authenticationMethod, "service");
+    // use the provided registry as the service (e.g., for OpenShift)
+    String service = repository;
```

Comment:
I believe the service should actually be the `serverUrl` (https://docs.docker.com/registry/spec/auth/token/#how-to-authenticate)

Reply:
Example: 

`curl -v https://gcr.io/v2/ 2>&1 >/dev/null | grep "www-authenticate"`

Reply:
The service will be replaced in the following two lines.  But let me rewrite it to be clearer.